modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/IgniteTransactionsImpl.java

```diff
@@ -103,8 +103,18 @@ public CompletableFuture<Transaction> beginAsync(@Nullable TransactionOptions op
         return CompletableFuture.completedFuture(begin(options));
     }
 
+    /**
+     * Begins a transaction.
+     *
+     * @param readOnly {@code true} in order to start a read-only transaction, {@code false} in order to start read-write one.
+     * @return The started transaction.
+     */
+    public InternalTransaction beginImplicit(boolean readOnly) {
```

Comment:
I believe it's not OK to add implicit flag to a transaction state as the part of this PR.
This introduces a log of unrelated bloating changes and better to be done as separate PR.

Reply:
The possibility to start an implicit transaction in another component is a main cause of this PR. Without this, we cannot handle an observable timestamp.

Reply:
All beginXXX methods alredy accept HybridTimestampTracker as an argument.
To fix the issue in the ticket description it's not necessary to add `implicit `state field to a transaction.